gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/dichotomy/shift/SweNetworkShifter.java

```diff
@@ -33,35 +33,35 @@
 public final class SweNetworkShifter implements NetworkShifter {
     private static final Logger LOGGER = LoggerFactory.getLogger(SweNetworkShifter.class);
     private static final double DEFAULT_SHIFT_EPSILON = 1;
-    private final ProcessConfiguration processConfiguration;
+    private static final int MAX_NUMBER_ITERATION = 20;
+    public static final String ES_PT = "ES_PT";
+    public static final String ES_FR = "ES_FR";
     private final Logger businessLogger;
 
     private final ProcessType processType;
     private final DichotomyDirection direction;
     private final ZonalData<Scalable> zonalScalable;
     private final ShiftDispatcher shiftDispatcher;
-    private final double toleranceEsPt;
-    private final double toleranceEsFr;
+    private final Tolerances tolerances;
     private final Map<String, Double> initialNetPositions;
+    private final ProcessConfiguration processConfiguration;
 
-    public SweNetworkShifter(Logger businessLogger, ProcessType processType, DichotomyDirection direction, ZonalData<Scalable> zonalScalable, ShiftDispatcher shiftDispatcher, double toleranceEsPt, double toleranceEsFr, Map<String, Double> initialNetPositions, ProcessConfiguration processConfiguration) {
+    public SweNetworkShifter(Logger businessLogger, ProcessType processType, DichotomyDirection direction, ZonalData<Scalable> zonalScalable, ShiftDispatcher shiftDispatcher, Tolerances tolerances, Map<String, Double> initialNetPositions, ProcessConfiguration processConfiguration) {
         this.businessLogger = businessLogger;
         this.processType = processType;
         this.direction = direction;
         this.zonalScalable = zonalScalable;
         this.shiftDispatcher = shiftDispatcher;
-        this.toleranceEsPt = toleranceEsPt;
-        this.toleranceEsFr = toleranceEsFr;
+        this.tolerances = tolerances;
         this.initialNetPositions = initialNetPositions;
         this.processConfiguration = processConfiguration;
     }
 
     @Override
     public void shiftNetwork(double stepValue, Network network) throws GlskLimitationException, ShiftingException {
         businessLogger.info("Starting shift on network {}", network.getVariantManager().getWorkingVariantId());
-
         Map<String, Double> scalingValuesByCountry = shiftDispatcher.dispatch(stepValue);
-        businessLogger.info(String.format("Target countries shift [ES = %.2f, FR = %.2f, PT = %.2f]", scalingValuesByCountry.get(toEic("ES")),  scalingValuesByCountry.get(toEic("FR")),  scalingValuesByCountry.get(toEic("PT"))));
```

Comment:
As it is a single comment, i would let it here (easier to read it immediately rather than jumping into a dedicated method)